tap_core/tests/manager_test.rs

```diff
@@ -204,6 +205,68 @@ async fn manager_create_rav_request_all_valid_receipts(
         .is_ok());
 }
 
+#[rstest]
+#[tokio::test]
+async fn deny_rav_due_to_wrong_value(
+    keys: (LocalWallet, Address),
+    allocation_ids: Vec<Address>,
+    domain_separator: Eip712Domain,
+    context: ContextFixture,
+) {
+    let ContextFixture {
+        context,
+        checks,
+        query_appraisals,
+        escrow_storage,
+        ..
+    } = context;
+    let manager = Manager::new(domain_separator.clone(), context, checks);
+    escrow_storage.write().unwrap().insert(keys.1, 999999);
+
+    let mut stored_signed_receipts = Vec::new();
+    for _ in 0..10 {
+        let value = 20u128;
+        let signed_receipt = EIP712SignedMessage::new(
+            &domain_separator,
+            Receipt::new(allocation_ids[0], value).unwrap(),
+            &keys.0,
+        )
+        .unwrap();
+        let query_id = signed_receipt.unique_hash();
+        stored_signed_receipts.push(signed_receipt.clone());
+        query_appraisals.write().unwrap().insert(query_id, value);
+        let _ = manager.verify_and_store_receipt(signed_receipt).await;
```

Comment:
Always handle/unwrap results unless there is a very good test-related reason.
```suggestion
        manager.verify_and_store_receipt(signed_receipt).await.unwrap();
```